controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -23,8 +23,10 @@ import (
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/docker/distribution/reference"
```

Comment:
wondering about the github.com/docker dependency..
it seems the package is fairly self-contained (only depending on go-digest), except that this still references a pretty large repository.

deferring to the maintainers, though.

one option is to fork this as the reference spec is [unlikely to change that much](https://github.com/docker/distribution/commits/master/reference/reference.go)

Reply:
This seems pretty in line with what is [being done in k/k](https://github.com/kubernetes/kubernetes/blob/d8695d06b7191db56ebbbc0340da263833c9bb6f/pkg/util/parsers/parsers.go), so I don't necessarily see an issue doing the same here (at least temporarily), might be good to have the k/k parser utilities put somewhere that is more friendly for vendoring, though :)

Reply:
Is there a downside to using plain string parsing to replace the tag and repository?

Reply:
a couple of downsides would be maintaining unit test coverage and code duplication across different repo utils. but at least the reference format is not going to change (ever).

we already have similar parsing in kubeadm and i can see @fabriziopandini is adding similar custom parsing for the cert manager image/tag support here:
https://github.com/kubernetes-sigs/cluster-api/pull/2558

Reply:
I tried looking real quick to see if there was any library provided by the OCI project for this purpose, since I would expect the image URL format to be part of the spec, but couldn't find anything in the brief digging that I did.

I also added a comment to @fabriziopandini's PR, since it doesn't take into account that there can be a ':' in the registry part of the URL, such as `my.docker.registry:5000/my/image:latest`

Reply:
@neolit123 - IIRC we were discussing at some point remove the docker/docker from k/k

If #2558 is introducing parcing, might as well make it reusable here

Reply:
though docker/docker (moby/moby now) is different from docker/distribution.
given docker/distribution is already in pkg/util/parsers i'd assume it's fine for now.